packages/ozone/src/team/index.ts

```diff
@@ -1,15 +1,22 @@
 import { Selectable } from 'kysely'
+import AtpAgent from '@atproto/api'
 import { chunkArray } from '@atproto/common'
 import { InvalidRequestError } from '@atproto/xrpc-server'
 import { AppContext } from '../context'
+import { DaemonContext } from '../daemon/context'
 import { Database } from '../db'
 import { Member } from '../db/schema/member'
 import { ids } from '../lexicon/lexicons'
 import { ProfileViewDetailed } from '../lexicon/types/app/bsky/actor/defs'
 import { Member as TeamMember } from '../lexicon/types/tools/ozone/team/defs'
 import { httpLogger } from '../logger'
+import { AuthHeaders } from '../mod-service/views'
 
 export type TeamServiceCreator = (db: Database) => TeamService
+export type AppviewCreator = {
```

Comment:
Refactoring away from passing the whole `AppContext` since we want to be able to run this from both app and daemon.

Reply:
Since you take the opportunity to refactor, you should change this from being a method  argument to a constructor argument:

```ts

class TeamService {
  constructor(
    public db: Database,
    protected readonly agent: AtpAgent,
    // Uses the same signature as existing "createAuthHeaders" methods
    protected readonly createAuthHeaders: (aud: string, lxm: string) => Promise<{ headers: Record<string, string> }>
  ) {}

  async getProfiles(dids: string[]) {
    // uses this.agent and this.createAuthHeaders
  }
}

```

Reply:
(this is the same comment as above)